modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/CommitWriteInvokeClosure.java

```diff
@@ -61,11 +68,23 @@ public void call(@Nullable VersionChain oldRow) throws IgniteInternalCheckedExce
             return;
         }
 
-        updateTimestampLink = oldRow.headLink();
-
         operationType = OperationType.PUT;
 
-        newRow = VersionChain.createCommitted(oldRow.rowId(), oldRow.headLink(), oldRow.nextLink());
+        RowVersion current = storage.readRowVersion(oldRow.headLink(), ALWAYS_LOAD_VALUE, false);
+        RowVersion next = oldRow.hasNextLink() ? storage.readRowVersion(oldRow.nextLink(), ALWAYS_LOAD_VALUE, false) : null;
+
+        // If the previous and current version are tombstones, then delete the current version.
+        if (next != null && current.isTombstone() && next.isTombstone()) {
+            toRemove = current;
+
+            newRow = VersionChain.createCommitted(oldRow.rowId(), next.link(), next.nextLink());
```

Comment:
Shouldn't newRow just match the oldRow, and the operation would be NOOP? Why would you need to PUT the same value?

Reply:
No, because we are updating links from `next` and not from the `current`.
Perhaps you will have a question about the code below on line 84, then we must execute `PUT` in order for `CommitWriteInvokeClosure#onUpdate` to be called.